### PR TITLE
[VL] Daily Update Velox Version (2024_06_04)

### PR DESCRIPTION
**Pull request by GlutenPerfBot**
Upstream Velox's New Commits:

```txt
61c33794d by Masha Basmanova, Fix error: comparison of unsigned expression in ‘< 0’ is always false [-Werror=type-limits]
c50a11708 by Sandino Flores, Add GCS dependencies to resolve_dependency and refactor cmake modules (9072)
f06e759ee by Masha Basmanova, Fix json_array_contains(json, NaN/Infinity/-Infinity) (10023)
c3ff3fce9 by Masha Basmanova, Eliminate try-catch from json_array_contains Presto function (10020)
061c02f81 by Ankita Victor, Update FileHandleGenerator to use file size and modification time (9466)
3d668970d by Masha Basmanova, Add inverse_normal_cdf Presto function (10022)
76d0a988e by Jimmy Lu, Optimize readWithVisitor for TrivialEncoding and MainlyConstantEncoding (10021)
5f7b6c48d by Jimmy Lu, Fix crash in RowContainer::storeWithNulls<TypeKind::HUGEINT>
579e8da8e by Jimmy Lu, Do not allocate dictionary filter cache if the filter is IsNotNull to avoid OOM (10001)
b9649f40e by yangshijie.2024, Fix CAST(short decimal as json) (10007)
894de1150 by Deepak Majeti, Disable CUDA/GPU build in CI until centos9 update (10025)
463d84b03 by Amit Dutta, Add map_remove_null_values Presto function (10009)
ab06a77a4 by Pramod, Add decimal mod operator (9351)
8648f50bb by Masha Basmanova, Reduce exception throwing in json_xxx Presto functions (10018)
43131a457 by Masha Basmanova, Add Further Optimizing TRY_CAST and TRY blog post (9998)
5efcaa446 by Zac Wen, Deprecate old ssd cache constructor (10003)
45b421dc0 by Masha Basmanova, Add json_array_get Presto function (10017)
7047e052f by Masha Basmanova, Refactor cast(x as date) (10010)
277d5c5e7 by Deepak Majeti, Add velox_type_tz to velox_config target (9995)
046ccd4f5 by Masha Basmanova, Fix parsing of 'March' and 'MARCH' input in date_parse Presto function (10005)
```

### CODE REVIEW

**Comment by acvictor** [comment]
@PHILO-HE the SsdCache constructor that Gluten was earlier using has been deprecated. Seems the values need to be moved to config.

**Comment by acvictor** [comment]
> @PHILO-HE the SsdCache constructor that Gluten was earlier using has been deprecated. Seems the values need to be moved to config.

https://github.com/facebookincubator/velox/blob/main/velox/common/caching/SsdCache.h#L32

File: dev/ci-velox-buildstatic.sh [hunk 1]
```diff
@@ -6,4 +6,4 @@ cd $GITHUB_WORKSPACE/
 source ./dev/vcpkg/env.sh
 sed -i '/^headers/d' ep/build-velox/build/velox_ep/CMakeLists.txt
 export NUM_THREADS=4
-./dev/builddeps-veloxbe.sh --build_tests=OFF --build_benchmarks=OFF --enable_s3=ON --enable_gcs=ON --enable_hdfs=ON --enable_abfs=ON
+./dev/builddeps-veloxbe.sh --build_tests=OFF --build_benchmarks=OFF --enable_s3=ON --enable_gcs=OFF --enable_hdfs=ON --enable_abfs=ON
```
Comment (rui-mo) [inline dev/ci-velox-buildstatic.sh:9]: Could you add this limitation in the PR description? Thanks.

**Review by rui-mo — APPROVED** [review]

**Comment by PHILO-HE** [comment]
Will fix static build issue when GCS is enabled in a follow-up pr.

**Pull request merged**